### DISCUSSION TITLE
Add context to Scopes tab within scope details

### DISCUSSION THREAD

**Discussion by JonSzeto821** [keycloak/keycloak#16709] [Archive]
Hello,

**Admin console section/page under feedback:** 
Admin console > Client scopes page > [one of the listed scopes] > Scopes tab

**Context:**
My team regularly works with customers for client enablements and managing scope configurations. Recently, we worked with a customer to get them enabled and configured with relevant scopes for their needs. The customer reported that certain claims were only showing up for some users - not all users (as intended). After some investigation within our team, it was unclear to us that this page added an effective permissions check before applying the client scope. 

**What we’d like to see/propose:**
We would ideally like to have some more context/direction around what should happen on this screen or how functionality on this tab should be utilized. Specifically information around client scope permissions: https://github.com/keycloak/keycloak-documentation/blob/main/server_admin/topics/clients/con-client-scopes.adoc#client-scopes-permissions

cc: @jsorah @vickeybrown @patrickjennings @adrhine

<img width="1680" alt="Screenshot 2023-01-30 at 2 26 59 PM" src="https://user-images.githubusercontent.com/31300416/215600175-f3a88f30-354f-4e93-9952-0f198acb6cd6.png">

**Comment by jonkoops** [comment]
@xianli123 perhaps you can see if this can be improved?

**Comment by xianli123** [comment]
Hi @jonkoops @JonSzeto821 . In my opinion, we can add an alert at the top of the detail page, as shown below. What do you think?

The context in the alert goes: "If there is no role scope mapping defined, each user is permitted to use this client scope. If there are role scope mappings defined, the user must be a member of at least one of the roles." Does it make sense? Thanks.

**Empty state**
<img width="896" alt="Screenshot 2023-01-31 at 13 30 54" src="https://user-images.githubusercontent.com/53931519/215674030-32c64d01-459d-4042-8199-ceefb7e438e2.png">

**Table view**
<img width="896" alt="Screenshot 2023-01-31 at 13 31 13" src="https://user-images.githubusercontent.com/53931519/215674075-b9575a73-fec0-440c-a947-a79304a9dc57.png">

In addition, user can hide this alert by the switch on the masthead.
<img width="643" alt="Screenshot 2023-01-31 at 13 45 31" src="https://user-images.githubusercontent.com/53931519/215676309-557c0ea1-aa03-40b4-92be-7ed60b05fdca.png">

**Reply by jonkoops** [reply]
Seems like a good solution to add a message. Not quite sure about the help mode, but we could implement it. @ssilvert @JonSzeto821 WDYT?

**Reply by adrhine** [reply]
I worked on this along with @JonSzeto821 , and my 2c are that that help message is super helpful. I wouldn't want to accidentally lose that message if I turned off help mode, so maybe I would just encourage everyone to not turn off help mode :grin:

**Reply by jonkoops** [reply]
The 'help mode' would be a new feature as well, perhaps we simply leave the message on screen at all times.

**Reply by adrhine** [reply]
That would work too. To us the behavior of this screen was easy to confuse with the "Client Scopes" tab on a Client, so making it as clear as possible that this page is doing something different (adding a permissions check basically), then it is awesome :+1:

**Reply by jonkoops** [reply]
I'd still like to have @ssilvert's take on this. As far as I am concerned we can create an issue for this.

**Comment by ssilvert** [comment]
I'll convert this to an issue.  

Just my opinion, but I think this message should go away if you turn help mode off.  The ability to turn off help mode is for super-users who want to save screen space.  This seems to be the perfect use case for it.

**Reply by JonSzeto821** [reply]
Thanks, @ssilvert ! Can you please provide us with a link to the issue once it gets created? We'd ideally like to include it in an RCA doc relating to this.

**Comment by ssilvert** [comment]
Hmm.  I thought the issue would be linked automatically.

@JonSzeto821 Here you go: https://github.com/keycloak/keycloak-ui/issues/4299
